wasm/jit/compiler.go

```diff
@@ -8,86 +8,311 @@ import (
+	// See wasm.OpcodeBr.
 	compileBr(o *wazeroir.OperationBr) error
+	// compileBrIf adds instrctions to pops a value and branch into ".then" lable if the value equals 1.
+	// Otherwise, the code branches into ".else" label.
 	compileBrIf(o *wazeroir.OperationBrIf) error
+	// compileBrTable adds instructions to do br_table operation.
+	// A br_table operation has list of targets and default target, and
+	// this pops a value from the stack (called "index") and decide which branch we go into next
+	// based on the value.
+	//
+	// For example, assume we have operations like {default: L_DEFAULT, targets: [L0, L1, L2]}.
+	// If "index" >= len(defaults), then branch into the L_DEFAULT label.
+	// Othewise, we enter label of targets[index].
+	// See wasm.OpcodeBrTable.
 	compileBrTable(o *wazeroir.OperationBrTable) error
+	// compileCall adds instructions to call into a funcion of the given index.
+	// See wasm.OpcodeCall.
 	compileCall(o *wazeroir.OperationCall) error
+	// compileCallIndirect adds instructions to perform call_indirect operation.
+	// This consumes the one value from the top of stack (called "offset"),
+	// and make a function call against the function whose function address equals "table[offset]".
+	//
+	// Note: This is called indirect function call in the sense that the target function is indirectly
+	// determined by the current state (top value) of the stack.
+	// Therefore, two checks are performed at runtime before entering the target function:
+	// 1) If "offset" exceeds the length of table, the function exits with jitCallStatusCodeInvalidTableAccess.
+	// 2) If the type of the function table[offset] doesn't match the specified function type, the function exits with jitCallStatusCodeTypeMismatchOnIndirectCall.
+	// Otherwise, we successfully enter the target function.
+	//
+	// Note: WebAssembly 1.0 (MVP) supports at most one table, so this doesn't support multiple tables.
+	// See wasm.CallIndirect.
 	compileCallIndirect(o *wazeroir.OperationCallIndirect) error
+	// compileDrop adds instructions to drop values within the given inclusive range from the value stack.
 	compileDrop(o *wazeroir.OperationDrop) error
+	// compileSelect uses top three values on the stack. For example, if we have stack as [..., x1, x2, c]
+	// and the value "c" equals zero, then the stack results in [..., x1], otherwise, [..., x2].
 	compileSelect() error
+	// compilePick adds instructions to copy a value on the given location in the Wasm value stack,
+	// and push the copied value onto the top of the stack.
 	compilePick(o *wazeroir.OperationPick) error
+	// compileAdd adds instructions to pop two values from the stack, add these two values, and push
+	// back the result onto the stack.
+	// See wasm.CallIndirect.
 	compileAdd(o *wazeroir.OperationAdd) error
+	// compileSub adds instructions to pop two values from the stack, subtract the top from the second one, and push
+	// back the result onto the stack.
+	// See wasm.Opcode{I32,I64,F32,F64}Sub.
```

Comment:
might as well expand so that reference search works
```suggestion
	// See wasm.OpcodeI32Sub wasm.OpcodeI64Sub wasm.OpcodeF32Sub wasm.OpcodeF64Sub
```